docs/CONTRIBUTING.md

```diff
@@ -32,3 +32,75 @@ Coming soon!
 After values are pushed the cluster's drone will do a deployment run. Helmfile diff is used targeting `$CLOUD` & `$CLUSTER` before the final deployment of the artifacts.
 
 If you have any meaningful additions to this, please let us know!
+
+## Schema validation
+
+### Guidelines/best practices
+
+We intend to be very exact in our speech regarding schema validation. Currently we perform meta-schema validation on the `values-schema.yaml` file. By having accurate and consistent descriptions (using the `description` keyword), we can provide the user with a better experience in his configuration management of the `otomi-values` repository.
+
+[Under the Apache V2.0 license agreement we are free to distribute the following content:](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#validation)
+
+API objects are validated upon receipt by the apiserver. Validation errors are flagged and returned to the caller in a Failure status with reason set to Invalid. In order to facilitate consistent error messages, we ask that validation logic adheres to the following guidelines whenever possible (though exceptional cases will exist).
+
+```
+- Be as precise as possible.
+- Telling users what they CAN do is more useful than telling them what they CANNOT do.
+- When asserting a requirement in the positive, use "must". Examples: "must be greater than 0", "must match regex '[a-z]+'". Words like "should" imply that the assertion is optional, and must be avoided.
+- When asserting a formatting requirement in the negative, use "must not". Example: "must not contain '..'". Words like "should not" imply that the assertion is optional, and must be avoided.
+- When asserting a behavioral requirement in the negative, use "may not". Examples: "may not be specified when otherField is empty", "only name may be specified".
+- When referencing a literal string value, indicate the literal in single-quotes. Example: "must not contain '..'".
+- When referencing another field name, indicate the name in back-quotes. Example: "must be greater than request".
+- When specifying inequalities, use words rather than symbols. Examples: "must be less than 256", "must be greater than or equal to 0". Do not use words like "larger than", "bigger than", "more than", "higher than", etc.
+- When specifying numeric ranges, use inclusive ranges when possible.
+```
+
+Excellent:
+
+```
+
+oauth2-proxy:
+type: object
+additionalProperties: false
+properties:
+config:
+type: object
+properties:
+cookieSecret:
+type: string
+description: Cookie secret must be 128 bit base64 encoded string.
+pattern: ^(?:[A-Za-z0-9+/]{4})\\\*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)?\$
```

Comment:
you are probably not aware that spaces were not copied here, and formatting does not work, so I conclude you have not looked at your copy paste action, which I hope you will in the future ;p

Reply:
Yes, thanks for noticing.